# Latihan Mata Kuliah Machine Learning - Pertemuan 6

## Import Library

In [ ]:
from google.colab import drive
import pandas as pd

## Mount Google Drive

In [ ]:
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


## Membaca Data



In [ ]:
path = '/content/drive/My Drive/Colab Notebooks/Dataset/house-prices-advanced-regression-techniques/train.csv'
train = pd.read_csv(path)

## Menampilkan 5 Baris Pertema

In [ ]:
print(train.head())

   Id  MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
0   1          60       RL         65.0     8450   Pave   NaN      Reg   
1   2          20       RL         80.0     9600   Pave   NaN      Reg   
2   3          60       RL         68.0    11250   Pave   NaN      IR1   
3   4          70       RL         60.0     9550   Pave   NaN      IR1   
4   5          60       RL         84.0    14260   Pave   NaN      IR1   

  LandContour Utilities  ... PoolArea PoolQC Fence MiscFeature MiscVal MoSold  \
0         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      2   
1         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      5   
2         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      9   
3         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      2   
4         Lvl    AllPub  ...        0    NaN   NaN         NaN       0     12   

  YrSold  SaleType  SaleCondition  SalePrice  
0   2008        WD   

## Cek Jumlah Baris dan Kolom (Baris, Kolom)

In [ ]:
print(train.shape)

(1460, 81)


## Lihat Nama-Nama Kolom

In [ ]:
print(train.columns)

Index(['Id', 'MSSubClass', 'MSZoning', 'LotFrontage', 'LotArea', 'Street',
       'Alley', 'LotShape', 'LandContour', 'Utilities', 'LotConfig',
       'LandSlope', 'Neighborhood', 'Condition1', 'Condition2', 'BldgType',
       'HouseStyle', 'OverallQual', 'OverallCond', 'YearBuilt', 'YearRemodAdd',
       'RoofStyle', 'RoofMatl', 'Exterior1st', 'Exterior2nd', 'MasVnrType',
       'MasVnrArea', 'ExterQual', 'ExterCond', 'Foundation', 'BsmtQual',
       'BsmtCond', 'BsmtExposure', 'BsmtFinType1', 'BsmtFinSF1',
       'BsmtFinType2', 'BsmtFinSF2', 'BsmtUnfSF', 'TotalBsmtSF', 'Heating',
       'HeatingQC', 'CentralAir', 'Electrical', '1stFlrSF', '2ndFlrSF',
       'LowQualFinSF', 'GrLivArea', 'BsmtFullBath', 'BsmtHalfBath', 'FullBath',
       'HalfBath', 'BedroomAbvGr', 'KitchenAbvGr', 'KitchenQual',
       'TotRmsAbvGrd', 'Functional', 'Fireplaces', 'FireplaceQu', 'GarageType',
       'GarageYrBlt', 'GarageFinish', 'GarageCars', 'GarageArea', 'GarageQual',
       'GarageCond', 'PavedDrive

## Cek Tipe Data dan Data Yang Kosong

In [ ]:
print(train.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 81 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Id             1460 non-null   int64  
 1   MSSubClass     1460 non-null   int64  
 2   MSZoning       1460 non-null   object 
 3   LotFrontage    1201 non-null   float64
 4   LotArea        1460 non-null   int64  
 5   Street         1460 non-null   object 
 6   Alley          91 non-null     object 
 7   LotShape       1460 non-null   object 
 8   LandContour    1460 non-null   object 
 9   Utilities      1460 non-null   object 
 10  LotConfig      1460 non-null   object 
 11  LandSlope      1460 non-null   object 
 12  Neighborhood   1460 non-null   object 
 13  Condition1     1460 non-null   object 
 14  Condition2     1460 non-null   object 
 15  BldgType       1460 non-null   object 
 16  HouseStyle     1460 non-null   object 
 17  OverallQual    1460 non-null   int64  
 18  OverallC

**Insight**

- **Jumlah baris**: 1460

- **Jumlah kolom**: 81

- **Tipe data**:

  - 3 kolom bertipe **float64** (misalnya `LotFrontage`, `MasVnrArea`).

  - 35 kolom bertipe **int64** (misalnya `Id`, `LotArea`, `OverallQual`).
  
  - 1 kolom bertipe **int8** (misalnya `Neighborhood`).

  - 42 kolom bertipe object (misalnya `MSZoning`, `Street`, `Alley`).

- **Kolom dengan nilai kosong**:

  - Beberapa kolom memiliki nilai kosong, seperti `LotFrontage` (259 missing), `Alley` (1370 missing), dan `MasVnrType` (872 missing).

## Lihat statistik data numerik

In [ ]:
print(train.describe())

                Id   MSSubClass  LotFrontage        LotArea  OverallQual  \
count  1460.000000  1460.000000  1201.000000    1460.000000  1460.000000   
mean    730.500000    56.897260    70.049958   10516.828082     6.099315   
std     421.610009    42.300571    24.284752    9981.264932     1.382997   
min       1.000000    20.000000    21.000000    1300.000000     1.000000   
25%     365.750000    20.000000    59.000000    7553.500000     5.000000   
50%     730.500000    50.000000    69.000000    9478.500000     6.000000   
75%    1095.250000    70.000000    80.000000   11601.500000     7.000000   
max    1460.000000   190.000000   313.000000  215245.000000    10.000000   

       OverallCond    YearBuilt  YearRemodAdd   MasVnrArea   BsmtFinSF1  ...  \
count  1460.000000  1460.000000   1460.000000  1452.000000  1460.000000  ...   
mean      5.575342  1971.267808   1984.865753   103.685262   443.639726  ...   
std       1.112799    30.202904     20.645407   181.066207   456.098091  ..

**Insight**

- **Jumlah data (count)**: Ada 1460 baris untuk hampir semua kolom, tapi beberapa kolom seperti `LotFrontage`, `MasVnrArea`, punya missing value (jumlahnya < 1460).

- **Rentang harga rumah** (`SalePrice`):

  - Rata-rata: **180,921**

  - Minimum: **34,900**

  - Maksimum: **755,000**

- **Luas tanah** (`LotArea`) sangat bervariasi:

  - Rata-rata: **10,516 m²**, tapi ada yang sampai **215,245 m²** (outlier).

- Tahun dibangun (`YearBuilt`):

  - Rumah tertua: **1872**, terbaru: **2010**

- **Beberapa fitur seperti** `WoodDeckSF`, `PoolArea`, `ScreenPorch` memiliki median = 0 → artinya banyak rumah yang **tidak punya** fitur tersebut.